### ISSUE TITLE
False deprecation warnings for escaped dollar sign in shell command

### ISSUE THREAD

**Issue by ksage** [ansible/ansible#4922]
This is a somewhat contrived example because the actual command I'm running is much longer/uglier, but if I have something like this:

```
- name: awk a file for no good reason
  shell: awk '{print \$2}' /some/file > /some/other/file
```

Then I'm getting a deprecation warning:

```
2013-11-15 00:48:54,976 p=32209 u=root |   [DEPRECATION WARNING]: Legacy variable subsitution, such as using ${foo} or
$foo instead of {{ foo }} is currently valid but will be phased out and has
been out of favor since version 1.2. This is the last of legacy features on our
deprecation list. You may continue to use this if you have specific needs for
now. This feature will be removed in version 1.6. Deprecation warnings can be
disabled by setting deprecation_warnings=False in ansible.cfg.
```

The shell command seems to work just fine, but it seems like the deprecation warning is because it's falsely detecting the escaped "\$2" in the shell command as an Ansible variable.

Note that it doesn't complain 100% of the time with this type of situation, but does so _most_ of the time.

I'm running from the tip of the devel branch as of this morning (de2f4cf04d).
Thanks!

**Comment by mpdehaan** [comment]
The code should be set to flag this when the string both changes the template and it does contain ${x} or $x but not "{{""

This is in queue to be reproduced, thanks!

**Comment by jctanner** [comment]
reproduced with ...

```
- hosts: localhost
  connection: local
  gather_facts: False
  tasks:
    - shell: echo "1 2" | awk '{print \$2}'
```

**Comment by mpdehaan** [comment]
Fix pushed in fc909b453f ... it deals with the \$ case only, but that should handle most false positives.
